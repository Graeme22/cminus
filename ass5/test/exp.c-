// C-F20
// C-F20
// C-F20
int g;

  main()
{
    {
        int x;
        int y;

        x=11;
        y=12;
        g=13;

        x=3+4*x+y*7+x*(y+g);
        x=(3-4/x-y/7-x/(y-g));
        x=((3%((4))/(x)%(y/(7)%x)/(y%g)));
    }
    {
        bool x, y, z[32];

        x=true;
        y=false;

        x=true | false & x | y & 7 | x & (y | g);
        x= ! (true) |  !  ! x;

        x = x  &  y  &  x;
        x = x  |  y  |  x;

        x=z[1];
        z[1]=x;
        z[1]=z[2];
        z[1]=z[2]+z[3]+*z;
    }
    {
        int z[3];

        x=z[1];
        x=?z[1];
        z[1]=x;
        z[1]=z[2];
        z[1]=z[2]+z[3];
        x=z[1]+z[2]*z[0]+z[3]*7+x*(z[4]+g);
    }

    {
         int a, b, c, d;
         a=b=c;
         a=b+c+d;
         a=b*c*d;
         g=a+=b-=c+d;
         a=(b=3)+(c=4);
    }
    {
         int a, b;
         bool x, y;
         x = x  |  y  &  b*a>a+b;
    }
}
