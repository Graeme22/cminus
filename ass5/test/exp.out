====================================
FILE: exp.c-
ERROR(26): '&' requires operands of type bool but rhs is of type int.
ERROR(26): '|' requires operands of type bool but rhs is of type int.
WARNING(32): Variable z may be uninitialized when used here.
ERROR(35): '+' requires operands of type int but lhs is of type bool.
ERROR(35): '+' requires operands of type int but rhs is of type bool.
ERROR(35): '=' requires operands of the same type but lhs is type bool and rhs is type int.
ERROR(40): Variable 'x' is not declared.
WARNING(40): Variable z may be uninitialized when used here.
ERROR(41): Variable 'x' is not declared.
ERROR(42): Variable 'x' is not declared.
ERROR(45): Variable 'x' is not declared.
ERROR(45): Variable 'x' is not declared.
WARNING(50): Variable c may be uninitialized when used here.
WARNING(51): Variable d may be uninitialized when used here.
WARNING(59): Variable x may be uninitialized when used here.
WARNING(59): Variable y may be uninitialized when used here.
WARNING(59): Variable b may be uninitialized when used here.
WARNING(59): Variable a may be uninitialized when used here.
Number of warnings: 8
Number of errors: 10
